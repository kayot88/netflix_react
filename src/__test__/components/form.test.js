import { render } from '@testing-library/react';
import React from 'react';
import Form from '../../components/form';

jest.mock('react-router-dom');

describe('<Form/>', () => {
  it('should render form with passing text ', () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign In</Form.Title>

        <Form.Base onSubmit={() => {}} method="POST">
          <Form.Input placeholder="Email address" onChange={() => {}} />
          <Form.Input
            type="password"
            placeholder="Password"
            onChange={() => {}}
          />
          <Form.Submit disabled type="submit">
            SignIn
          </Form.Submit>
        </Form.Base>

        <Form.Text>
          New to Netflix? <Form.Link to="/signup">Sign up now.</Form.Link>
        </Form.Text>
        <Form.TextSmall>
          This page is protected by Google reCAPTCHA to ensure you're not a bot.
          Learn more.
        </Form.TextSmall>
      </Form>
    );
    expect(getByText('Sign In')).toBeTruthy();
    expect(getByText('New to Netflix?')).toBeTruthy();
    expect(getByText('SignIn').disabled).toBeTruthy();
    expect(getByPlaceholderText('Password')).toBeTruthy();
    expect(getByPlaceholderText('Email address')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('form with an error ', () => {
    const { container, getByText, queryByText } = render(
      <Form>
        <Form.Error>
          The email address is already in use by another account.
        </Form.Error>
        <Form.Submit type="submit">Sign In</Form.Submit>
      </Form>
    );
    expect(
      getByText('The email address is already in use by another account.')
    ).toBeTruthy();
    expect(queryByText('Sign In').disabled).toBeFalsy();
  });
});
