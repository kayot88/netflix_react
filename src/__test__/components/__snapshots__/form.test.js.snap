// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form/> form with an error  1`] = `undefined`;

exports[`<Form/> should render form with passing text  1`] = `
<div
  class="sc-bdfBwQ gFJsfN"
>
  <h1
    class="sc-hKgILt gPncSF"
  >
    Sign In
  </h1>
  <form
    class="sc-dlfnbm gsUiWl"
    method="POST"
  >
    <input
      class="sc-iBPRYJ hfOyNP"
      placeholder="Email address"
    />
    <input
      class="sc-iBPRYJ hfOyNP"
      placeholder="Password"
      type="password"
    />
    <button
      class="sc-fubCfw fXpolG"
      disabled=""
      type="submit"
    >
      SignIn
    </button>
  </form>
  <h1
    class="sc-eCssSg fhMwce"
  >
    New to Netflix? 
  </h1>
  <h1
    class="sc-jSgupP cjRwrH"
  >
    This page is protected by Google reCAPTCHA to ensure you're not a bot. Learn more.
  </h1>
</div>
`;
